obspy/io/sac/tests/test_sactrace.py

```diff
@@ -433,6 +433,33 @@ def test_data_headers(self):
                            ('depmax', depmax), ('npts', npts)]:
             self.assertEqual(getattr(sac, hdr), value)
 
+    def test_char_header_padding(self):
+        """
+        SAC binary file header fields should be padded with blanks, not NULs,
+        or NUL-terminated.
+        """
+
+        dat = np.zeros(8)
+
+        sac = SACTrace(
+            kstnm='TEST',
+            delta=1,
+            kcmpnm='Z',
+            kevnm='Bug test',
+            npts=len(dat),
+            data=dat
+        )
+        with NamedTemporaryFile() as tf:
```

Comment:
You need to delete this file after, but do you really need a file? Can you use a `io.BytesIO` instead?

Reply:
Yep, this should be BytesIO

Reply:
actually we have an `os.remove(...)` in `__exit__` but still better BytesIO

Reply:
Yea, we have `__exit__`, but this is using the old trick of creating a temporary file to actually just come up with a _name_, and then working with that. The delete from the context manager happens before the file is actually written.